gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/PolygonDrawingViewModel.java

```diff
@@ -185,6 +188,13 @@ private void addVertex(Point vertex, boolean isNotSelectedByUser) {
     updateDrawnPolygonFeature(ImmutableList.copyOf(vertices));
   }
 
+  private String getId() {
+    if (uuid == null) {
+      uuid = uuidGenerator.generateUuid();
```

Comment:
I'm not 100% sure since the code gets invoked using RxJava and has multiple operators in between. But, it should be mostly thread safe since the events downstream are serialized and sent sequentially. Also, there is only 1 subject that writes to that stream. Also, this code would be further refactored to move UUID generation at the beginning of polygon drawing in later PRs. Thanks!